src/index.js

```diff
@@ -79,6 +78,33 @@ async function start(fields, doRetry = true) {
       googleUtils
     )
 
+    // update the contact account
+    const lastLocalSync = new Date().toISOString()
+    let nextSyncToken = syncTokenBefore
+    // if we changed something on google, we have to ask for a new sync token
+    if (Object.values(result.google).some(v => v !== 0)) {
+      const response = await googleUtils.getAllContacts({
+        syncToken: syncTokenBefore
+      })
+      nextSyncToken = response.nextSyncToken
+      const contactsWeUpdated = result.google.created + result.google.updated
+      if (response.contacts.length !== contactsWeUpdated) {
+        log(
+          'warn',
+          'User has created/updated contacts on google during the synchronization ' +
+            '(some cozy changes may be lost next time).'
+        )
```

Comment:
In that case I would also not use the `response.nextSyncToken`, since using it will actively throw away changes made on google ?

Reply:
if we do not use it, we use the previous one and so we could lost more data during the next sync whereas if we use it, we will lose data only if we edit on cozy the contacts that we edited on google (in this case, cozy changes will be overiden by google changes). Not a big deal, especially since it can happen only if we edit those same contacts on cozy until next run of the connector (i.e. the next night)

Reply:
Ah ok, I guess I had it backwards then. All good for me!